great_expectations/checkpoint/checkpoint.py

```diff
@@ -76,6 +76,25 @@ def __init__(
             raise TypeError("A Checkpoint requires a valid DataContext")
         self._data_context = data_context
 
+        # if checkpoint was created without yml
+        if validations:
+            for idx, val in enumerate(validations):
+                if "batch_request" in val:
+                    if isinstance(val["batch_request"], RuntimeBatchRequest):
+                        batch_data = validations[idx]["batch_request"][
+                            "runtime_parameters"
+                        ]["batch_data"]
```

Comment:
@NathanFarmer I have a couple of questions about the line `batch_data = validations[idx]["batch_request"]["runtime_parameters"]["batch_data"]`:
* Since `"batch_request" in val` and `val = validations[idx]`, could we simplify the above line to say `batch_data = val["batch_request"]["runtime_parameters"]["batch_data"]`?
* If the above is correct, could we then change the next line to `val["batch_request"]["runtime_parameters"]["batch_data"] = batch_data`?  This should modify `validations` successfully, because `val` is a reference.
* If both of the above are correct, then perhaps we could simplify the iteration to say `for val in validations`?

Thank you! /cc @anthonyburdi

Reply:
✅ Like you stated, since `val` is a reference all of these changes were made.